src/platform/emotes/data.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  generateValidator,
+  JSONSchema,
+  ValidateFunction
+} from '../../validation'
+
+/** @alpha */
+export type EmoteData = {
+  loop: boolean
+  thumbnail: string
+}
+
+/** @alpha */
+export namespace EmoteData {
+  export const schema: JSONSchema<EmoteData> = {
+    type: 'object',
+    properties: {
+      loop: {
+        type: 'boolean'
+      },
+      thumbnail: {
+        type: 'string',
+        minLength: 1
+      }
+    },
+    additionalProperties: false,
+    required: ['loop', 'thumbnail']
+  }
+
+  export const nullableSchema: JSONSchema<EmoteData> = {
```

Comment:
In the `Emote` the `data` is required but in the `EmoteRepresentation` is not. I thought it would add clarity to have it explicitly declared as `nullable`, but there's no strong need on our side to receive `null` `data`, we can remove it.